src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -206,6 +206,7 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_predicate_node_to_in
   // TableScan(s).
   auto index_scan = std::make_shared<IndexScan>(input_operator, SegmentIndexType::GroupKey, column_ids,
                                                 predicate->predicate_condition, right_values, right_values2);
+  index_scan->set_lqp_node(node);
```

Comment:
The support in the LQPTranslator looks incomplete. Any reason not to make this a part of `LQPTranslator::translate_node` and add an exception for the index scan (which will be redundant with #1790)?

Reply:
The reason is (and I don't know if that's really an issue though) that we do not want to have the overhead of keeping LQP nodes around for each and every single operator even though it is never accessed for later analyses.
Currently, only operators that we analyse for predictions are storing their LQP nodes.

Reply:
Having support only for whatever you considered worthwhile for your current work seems inconsistent.

Reply:
Hmm ... looking at how much we "save" right now it not worth the differentiation anyways. Will change.

Reply:
Done.